joblib/numpy_pickle.py

```diff
@@ -333,44 +338,32 @@ def load_build(self):
         dispatch[pickle.BUILD] = load_build
 
 
-class ZipNumpyUnpickler(NumpyUnpickler):
-    """A subclass of our Unpickler to unpickle on the fly from
-    compressed storage."""
-
-    def __init__(self, filename, file_handle):
-        NumpyUnpickler.__init__(self, filename,
-                                file_handle,
-                                mmap_mode=None)
-
-    def _open_pickle(self, file_handle):
-        return BytesIO(read_zfile(file_handle))
-
-
 ###############################################################################
 # Utility functions
 
-def dump(value, filename, compress=0, cache_size=100, protocol=None):
-    """Fast persistence of an arbitrary Python object into one or multiple
-    files, with dedicated storage for numpy arrays.
+def dump(value, filename, compress=0, protocol=None, cache_size=None):
+    """Persist an arbitrary Python object into one file.
 
     Parameters
     -----------
     value: any Python object
         The object to store to disk
-    filename: string or pathlib.Path
+    filename: str or pathlib.Path
         The path of the file in which it is to be stored
```

Comment:
I like the idea and I just pushed a commit with the change. The documentation still has to be updated though.

Just one comment about this feature: the compression level is now set automatically to 3 which is not that nice I think. Maybe there's a way to use both file extension and `compress`.